### ISSUE TITLE
.scubainit breaks everything on OSX

### ISSUE THREAD

**Issue by JonathonReinhart** [JonathonReinhart/scuba#17]
Trying the latest (742eccf5) on OSX with boot2docker doesn't work, at all. It just quietly exits with status 0:
```
$ scuba
$ echo $?
0
```

Commenting out the tempfile removal, and running the command that scuba would run, docker does the same thing, so Scuba is at least not "hiding" the error.

**Comment by JonathonReinhart** [comment]
On OXX, the scubainit temp file is created in `/var/folders/...`.   When attempting to bind-mount this into the container, there ends up being *an empty directory* mounted at `.scubainit`, instead of the file contents.

I would have expected `/bin/sh .scubainit_which_is_a_directory` to have at least printed an error, like:
```
$ /bin/sh /tmp
/tmp: /tmp: is a directory
```
but... it didn't.

[This StackOverflow answer](http://stackoverflow.com/a/26549438/119527) indicates that Docker 1.3 enabled host mounts on OSX, but only from `/Users`:

> With this release we are addressing the most common issue: sharing directories between your Mac and your containers. Using Docker 1.3 with the corresponding version of boot2docker, host-mounted volumes now work the way you expect them to.

> ...Note that there are still some limitations: for example this feature is limited to boot2docker’s virtualbox configuration, cannot be managed dynamically, and only works for directories in /Users . But we are receiving exciting contributions to improve volume management, so expect this area to improve drastically in the next few releases.

Here's at least one related issue: docker/machine/issues/2736

**Comment by JonathonReinhart** [comment]
Additionally, it seems that all of the user creation and `su` in `.scubainit` is problematic when using boot2docker.  For example, the default `staff` group is gid 20, which conflicts with system groups in Linux.  There's no need for any of that on OSX, where, because of the shared folder implementation, all files are created on the host owned as the correct user anyway.

So we've got two options:

1. If `DOCKER_HOST` is set (implying remote boot2docker), forgo the entire `.scubainit` business, and just execute the command as we did in v1.2.0.  This has the slight disadvantage that if we add anything else interesting to `.scubainit`, it will not work on OSX.

2. Create the temporary scubainit file somewhere in `/Users`, (probably in the current directory) so it can be mounted into the container. Then, omit the user creation and `su` from `.scubainit` when `DOCKER_HOST` is set.
